pandas/core/dtypes/cast.py

```diff
@@ -148,6 +149,30 @@ def maybe_box_datetimelike(value: Scalar, dtype: Optional[Dtype] = None) -> Scal
     return value
 
 
+def maybe_box_native(value: Scalar) -> Scalar:
```

Comment:
can you add some unit tests for this function

Reply:
Done (in `pandas/tests/dtypes/cast/test_maybe_box_native.py`)